### PR TITLE
feat(Stamps): update styling of stamps component

### PR DESCRIPTION
**Pull request by Sarah4VT**
#### Summary

Updating the styling of the stamps component to reflect new designs created for Audit Log.

## Description

[Design](https://www.figma.com/file/v6TLyh6Egyh5NJxWOb3ixK/New-Audit-Log?type=design&node-id=241-8228&mode=design&t=pPZdqNKV8VQWlnDA-4)
image

Here are how the new stamps look:
Regular 
image

Condensed 
image

### CODE REVIEW

**Comment by Sarah4VT** [comment]
Putting this as WIP because I imagine the Percy screenshots will need to be updated for the visual tests, but I've never done that before and am not sure I have the right setup yet. Will probably need some help figuring that out.

File: design-system/materials/internals/definition.yaml
Comment (CarlosCortizasCT) [inline design-system/materials/internals/definition.yaml:65]: Maybe we should be creating a new token here (and the same for warning and error tokens) since the name is following some implicit convention and this would change not only `Stamp` component component but any other which uses this token.
In `ui-kit` there's only this component using this token, but it's an exported one so any other external consumer would have this color change in whatever context they're using it.

File: design-system/materials/internals/definition.yaml
Comment (CarlosCortizasCT) [inline design-system/materials/internals/definition.yaml:1118]: @FilPob Do you think we will be using this `2px` value elsewhere? If so, maybe it make sense to create a new design token like `spacing-05` or something like that instead.

**Comment by CarlosCortizasCT** [comment]
> Putting this as WIP because I imagine the Percy screenshots will need to be updated for the visual tests, but I've never done that before and am not sure I have the right setup yet. Will probably need some help figuring that out.
> 
> @commercetools/shield-team-fe Is there someone that can help me with this?

Hi Sarah ✋ 

Thanks for contributing to `ui-kit` 🙇 

Regarding the Percy screenshots, what we usually do is to add @FilPob as a reviewer so he can double check the new UI is right.
After he does his review, he will approve the changes in Percy and we will have that check passing in the PR.

**Comment by CarlosCortizasCT** [comment]
One thing I noticed is that the text styles don't seem to match the expected values when I compare them with the ones in Figma:
Screenshot 2023-10-02 at 09 31 03

`Stamp` component is currently using the `Text.Detail` component which has this font styles:
* size: `14px`
* weight: `400` | `600`

So it seems the new styles don't align with that and maybe we need to not use `Text.Detail` component anymore and have a custom text styles for this component.

cc/ @FilPob

**Review by CarlosCortizasCT — APPROVED** [review]

**Comment by FilPob** [comment]
> One thing I noticed is that the text styles don't seem to match the expected values when I compare them with the ones in Figma: Screenshot 2023-10-02 at 09 31 03
> 
> `Stamp` component is currently using the `Text.Detail` component which has this font styles:
> 
> * size: `14px`
> * weight: `400` | `600`
> 
> So it seems the new styles don't align with that and maybe we need to not use `Text.Detail` component anymore and have a custom text styles for this component.
> 
> cc/ @FilPob

There is one more change that needs to be done prio to this task, and that is the inclusion of a new text component (with 12px font-size) see requirements [here](https://www.figma.com/file/v6TLyh6Egyh5NJxWOb3ixK/New-Audit-Log?type=design&node-id=200%3A5936&mode=design&t=dq5ciuqOjXboz3yA-1)

File: design-system/materials/internals/definition.yaml
Comment (Sarah4VT) [reply design-system/materials/internals/definition.yaml:65]: I'm holding off on this PR until the Color and Typography stories get completed. @valoriecarli  is working on the color story and can get the colors defined as we will need them for stamps.

So it sounds like you are asking that we go back to adding new <color>-35 tokens for all the stamp colors?

**Comment by emmenko** [comment]
@FilPob Can you review the VRTs?

**Comment by FilPob** [comment]
@emmenko there is something weird because in the VRT I see that the font-weight is already correct (500) (at least looks like it in the screenshots)  but in the preview link in storybook I still see 400.

**Comment by Sarah4VT** [comment]
> @emmenko there is something weird because in the VRT I see that the font-weight is already correct (500) (at least looks like it in the screenshots) but in the preview link in storybook I still see 400.

Sorry I labeled this as WIP but forgot to put it as draft. I'm waiting for the Typography stuff to finish and then I will make a few more changes to this PR, which I have locally, but haven't tested with Jon's Typography changes yet since there is still work being done there (or there was yesterday).

File: design-system/materials/internals/definition.yaml
Comment (Sarah4VT) [reply design-system/materials/internals/definition.yaml]: This PR no longer changes colors of the stamps as specified in the updated Figma.

**Comment by Sarah4VT** [comment]
> > One thing I noticed is that the text styles don't seem to match the expected values when I compare them with the ones in Figma: Screenshot 2023-10-02 at 09 31 03
> > `Stamp` component is currently using the `Text.Detail` component which has this font styles:
> > 
> > * size: `14px`
> > * weight: `400` | `600`
> > 
> > So it seems the new styles don't align with that and maybe we need to not use `Text.Detail` component anymore and have a custom text styles for this component.
> > cc/ @FilPob
> 
> There is one more change that needs to be done prio to this task, and that is the inclusion of a new text component (with 12px font-size) see requirements [here](https://www.figma.com/file/v6TLyh6Egyh5NJxWOb3ixK/New-Audit-Log?type=design&node-id=200%3A5936&mode=design&t=dq5ciuqOjXboz3yA-1)

For the condensed scenario, I used the <Text.Caption> component with font-weight="medium" and it resolves to a font size of 12.8px.

**Comment by Sarah4VT** [comment]
@CarlosCortizasCT @FilPob @emmenko This is ready for review now.

**Comment by CarlosCortizasCT** [comment]
Something I noticed when checking the condensed version is that the styles for the `Text.Caption` component do not match the ones in the original Figma file.
It is supposed to have:
* font-size: 12px (0.75rem)
* line-heignt: 18px (1.125rem)

but currently the component has these values:
* font-size: 0.8rem (12.8px)
* line-height: 1.3rem (20.8px)

image

**Comment by FilPob** [comment]
Besides Carlos comment I would have one more color adjustment wish. To change the background color for tones `primary` and `positive` from color-primary-`95` to color-primary-`90` Otherwise looks awesome :)

**Comment by FilPob** [comment]
@Sarah4VT oh true. Can you please add that color? The value should be `hsl(175, 70%, 90%)`

**Comment by CarlosCortizasCT** [comment]
> @CarlosCortizasCT I can change `font-size-for-text-as-caption` to use `font-size-10` which has a value of `0.75rem` rather than `font-size-12` which has a value of `0.8rem`.
> 
> However, for line height, there is no line-height choice with the value of 1.125. Here are the current choices:
> 
> ```
>         line-height-10: '1.25rem'
>         line-height-18: '1.3rem'
>         line-height-20: '1.375rem'
>         line-height-30: '1.5rem'
>         line-height-40: '1.625rem'
>         line-height-50: '1.75rem'
>         line-height-60: '2.125rem'
> ```
> 
> Do you want me to add a new choice with this value or change an existing one?

Yes, I think it makes sense to include a new choice token in this case:
```
line-height-05: '1.125rem'
```

Thanks!

**Comment by Sarah4VT** [comment]
> > @CarlosCortizasCT I can change `font-size-for-text-as-caption` to use `font-size-10` which has a value of `0.75rem` rather than `font-size-12` which has a value of `0.8rem`.
> > However, for line height, there is no line-height choice with the value of 1.125. Here are the current choices:
> > ```
> >         line-height-10: '1.25rem'
> >         line-height-18: '1.3rem'
> >         line-height-20: '1.375rem'
> >         line-height-30: '1.5rem'
> >         line-height-40: '1.625rem'
> >         line-height-50: '1.75rem'
> >         line-height-60: '2.125rem'
> > ```
> > 
> > 
> > Do you want me to add a new choice with this value or change an existing one?
> 
> Yes, I think it makes sense to include a new choice token in this case:
> 
> ```
> line-height-05: '1.125rem'
> ```
> 
> Thanks!

This is complete now and screenshots updated in the description. Thanks!

**Comment by FilPob** [comment]
@Sarah4VT  looks good from my side now, thanks :)

**Comment by Sarah4VT** [comment]
> @Sarah4VT looks good from my side now, thanks :)

Great! Mind hitting the approve button so it's easy to see?

**Review by culotta — APPROVED** [review]

**Review by CarlosCortizasCT — COMMENTED** [review]
Looks good from my side 👍 

I'll approve once the merge conflicts are resolved.

**Pull request merged**
